packages/@expo/cli/src/utils/codesigning.ts

```diff
@@ -348,10 +348,13 @@ function validateStoredDevelopmentExpoRootCertificateCodeSigningInfo(
   );
 
   // TODO(wschurman): maybe move to @expo/code-signing-certificates
-  const leafCertificate = certificateChain[0];
-  const now = new Date();
-  if (leafCertificate.validity.notBefore > now || leafCertificate.validity.notAfter < now) {
-    return null;
+
+  // ensure all intermediate certificates are valid (in case we forget to rotate)
```

Comment:
Not just forgetting to rotate, but if the signing certificate (the leaf) expires after the intermediate certificate, too, right?

Say the leaf expires on March 15 and the intermediate certificate expires on February 28. We rotate the intermediate certificate on February 27, but the client CLI uses the old intermediate certificate while it's in offline mode. You could make the argument that the leaf should not have such an expiration (feature for another PR?) but in this case, validating the intermediate certificate would be necessary.

Reply:
Theoretically if we rotate every year this will never be the case, but yeah, I agree. I thought about that (setting the leaf expiration to `MIN(+30 days, intemediate expiration)`) but in the end the result would be the same (invalid chain).

Reply:
I think it's still a good idea to have the child's expiration not exceed its parent's - it's too easy (like in this scenario) to look at just the child and see it's unexpired and to think it's valid, but not a necessary constraint to implement.